hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneTenantShell.java

```diff
@@ -816,6 +818,22 @@ public void testTenantSetSecret() throws IOException, InterruptedException {
       return null;
     });
 
+    // Assigning/revoking user accessIds should work for a non-delegated admin
```

Comment:
I think we should split this off into its own test method. In that method, we can have a delegated and non-delegated admin. For each admin operation (add/revoke user, set secret, add/revoke admin) we try it as the delegated and non-delegated admin, and assert they pass or fail as expected. This should cover the full matrix of possible admin operations to admin type combinations.

Reply:
I agree. Some of the JUnit test methods in this class has grown too large. Should probably do a test refactoring at some point (at the cost of slightly longer integration test run time due to the extra init/cleanup, but should be worth it).

Moved the test to a **new** JUnit test method for assigning/revoking tenant accessIds (users) as non-delegated admin.

Reply:
I have made it a new method `testTenantAdminOperations`. Should be better now.

I'm keeping `testTenantSetSecret` separate as it is testing some edge cases (e.g. set empty secret) that we don't really care in `testTenantAdminOperations`, albeit overlapping slightly with the test addition here.